File: basics/string-comparison/python/string-comparison.py
import sys

def main():
    s1 = sys.argv[1]
    s2 = sys.argv[2]
    if s1 == s2: print('\'%s\' is equal to \'%s\' with case.' % (s1, s2))
    if s1.lower() == s2.lower(): print('\'%s\' is equal to \'%s\' without case.' % (s1, s2))
    if s1 < s2: print('\'%s\' is lexically before \'%s\'.' % (s1, s2))
    if s1 == s2: print('\'%s\' is lexically equal to \'%s\'.' % (s1, s2))
    if s1 > s2: print('\'%s\' is lexically after \'%s\'.' % (s1, s2))

if __name__ == "__main__":
    main()

File: basics/string-comparison/ruby/StringComparison.rb
s1 = ARGV[0]
s2 = ARGV[1]

if (s1 == s2) then
	puts("'#{s1}' is equal to '#{s2}' with case.");
end
if (s1.downcase == s2.downcase) then
	puts("'#{s1}' is equal to '#{s2}' without case.");
end

if (s1 < s2) then
	puts("'#{s1}' is lexically before '#{s2}'.");
end
if (s1 == s2) then
	puts("'#{s1}' is lexically equal to '#{s2}'.");
end
if (s1 > s2) then
	puts("'#{s1}' is lexically after '#{s2}'.");
end